openedx_proversity_reports/middleware.py

```diff
@@ -0,0 +1,98 @@
+"""
+Middlewares for openedx-proversity-reports.
+"""
+import json
+from datetime import datetime, timedelta
+
+from django.conf import settings
+from openedx_proversity_reports.edxapp_wrapper.get_student_library import get_user_profile
+
+
+class UserSessionMiddleware(object):
+    """
+    Middleware to store session data.
+    """
+    AVG_SESSION_LABEL = 'avg_session'
+    LAST_SESSION_LABEL = 'last_session'
+    LENGTH_CURRENT_SESSION_LABEL = 'length_current_session'
+    SESSION_NUMBER_LABEL = 'session_number'
+    TIME_BETWEEN_SESSIONS_LABEL = 'time_between_sessions'
+
+    def process_request(self, request):
+        """
+        This middleware stores the session date and calculates the
+        average session time, average time between sessions and current length
+        session for every authenticated user.
+        """
+        user = request.user
+
+        if user.is_anonymous():
+            return None
+
+        user_profile = get_user_profile().objects.get(user_id=user.id)
+
+        try:
+            meta = json.loads(user_profile.meta)
+            last_session = meta.get(self.LAST_SESSION_LABEL)
+            last_session = datetime.strptime(last_session, '%Y-%m-%d %H:%M:%S.%f')
+            avg_session = float(meta.get(self.AVG_SESSION_LABEL, 0))
+            length_current_session = float(meta.get(self.LENGTH_CURRENT_SESSION_LABEL, 0))
+            time_between_sessions = float(meta.get(self.TIME_BETWEEN_SESSIONS_LABEL, 0))
+            session_number = int(meta.get(self.SESSION_NUMBER_LABEL, 0))
```

Comment:
Isn't better to do these conversion from string to float/int in a Serializer? @andrey-canon